.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: ci
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+      with:
+        fetch-depth: 0
+
+    - name: ci/setup-go
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+      with:
+        go-version: "1.19"
+        cache: true
+
+    - name: ci/verify-gomod
+      run: make verify-gomod
+
+    - name: ci/check-style
+      run: make check-style
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+
+    - name: ci/setup-go
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+      with:
+        go-version: "1.19"
+        cache: true
+
+    - name: ci/test
+      run: |
+        export PATH=$(go env GOPATH)/bin:$PATH
+        make unittest
+
+  build:
+    if: ${{ github.event_name == 'pull_request' || github.ref_name  == 'master' }}
+    permissions:
+      security-events: write
+    runs-on: ubuntu-latest
+    needs: [lint, test]
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@dc323e67f16fb5f7663d20ff7941f27f5809e9b6 # v3.2.0
+      with:
+        fetch-depth: 0
+
+    - name: ci/set-short-SHA
+      run: echo "SHORT_SHA=${GITHUB_SHA:0:7}" >> $GITHUB_ENV
+
+    - name: ci/build-docker
+      run: make build-image
+
+    - name: ci/scan-docker-security
+      uses: aquasecurity/trivy-action@8bd2f9fbda2109502356ff8a6a89da55b1ead252 # v0.9.1
+      continue-on-error: true
+      with:
+        image-ref: "mattermost/bifrost:test"
+        format: "sarif"
+        limit-severities-for-sarif: true # https://github.com/aquasecurity/trivy-action/blob/0.9.1/entrypoint.sh#L172
+        output: "trivy-results.sarif"
+        exit-code: "0"
+        ignore-unfixed: true
+        vuln-type: "os,library"
+        severity: "CRITICAL,HIGH"
+
+    - name: ci/create-trivy-results-report
+      # if: failure()
+      uses: github/codeql-action/upload-sarif@17573ee1cc1b9d061760f3a006fc4aac4f944fd5 # v2.2.4
+      with:
+        sarif_file: "trivy-results.sarif"
+
+    - name: ci/setup-buildx
+      uses: docker/setup-buildx-action@15c905b16b06416d2086efa066dd8e3a35cc7f98 # v2.4.0
+      with:
+        version: v0.7.1
+
+    - name: ci/docker-login
+      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a # v2.1.0
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
Heads up. These secrets are not getting inherited to PRs raised from Forks. Therefore all forks CI will fail every-time in this step. 


Since this is a public repository, consider splitting this docker-push and docker-login steps to a seperate workflow, that get's trigger with `workflow_run` trigger. 
Example: https://github.com/mattermost/mattermost-server/blob/master/.github/workflows/artifacts.yml